sample_structure.py

```diff
@@ -79,10 +77,17 @@ def load_and_write_surfaces(self):
         for bulk_ind, bulk in enumerate(self.all_bulks):
             possible_surfaces = bulk.get_possible_surfaces()
             if self.args.enumerate_all_structures:
-                self.logger.info(f'Enumerating all {len(possible_surfaces)} surfaces for bulk {self.bulk_indices_list[bulk_ind]}')
-                for surface_ind, surface_info in enumerate(possible_surfaces):
-                    surface = Surface(bulk, surface_info, surface_ind, len(possible_surfaces))
-                    self.combine_and_write(surface, self.bulk_indices_list[bulk_ind], surface_ind)
+                if self.args.surface_index is not None:
+                    self.logger.info(f'Loading only surface {self.args.surface_index} for bulk {self.bulk_indices_list[bulk_ind]}')
+                    included_surface_indices = [self.args.surface_index]
```

Comment:
probably need to do a try/except clause here for making sure that the surface_index is within the appropriate bounds.